app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -64,6 +64,8 @@ def find_user(auth_type)
 
     if request['media_object_id']
       redirect_to media_object_path(request['media_object_id'])
+    elsif request['section_id']
+      redirect_to masterfile_path(request['section_id'])
```

Comment:
What if we replace both `media_object_id` and `section_id` with `target_id` and figure out which class type to route to on dispatch? That might even let us create a nicer route (say, `/courselink/:target_id`, that routes to `users/omniauth_callbacks {:action=>"lti",:target_id=>target_id}`)